libs/agno/agno/tools/evm.py

```diff
@@ -0,0 +1,96 @@
+"""EVM integration tools for interacting with EVM Blockchains."""
+
+from os import getenv
+
+from web3 import Web3
+
+from agno.tools import Toolkit
+from agno.utils.log import logger
+
+
+class EvmTools(Toolkit):
+    def __init__(
+        self,
+        private_key: str,
+        rpc_url: str,
+    ):
+        """Initialise EVM tools."""
+
+        super().__init__(name="evm_tools")
+
+        self.private_key = private_key or getenv("EVM_PRIVATE_KEY")
+        self.rpc_url = rpc_url or getenv("EVM_RPC_URL")
+
+        if not self.private_key:
+            logger.error("Private Key is required")
+            raise ValueError("Private Key is required")
+        if not self.rpc_url:
+            logger.error("RPC Url is needed to interact with EVM blockchain")
+            raise ValueError("RPC Url is needed to interact with EVM blockchain")
+        if not self.private_key.startswith("0x"):
```

Comment:
Why this? Shouldn't we just expect the correct key?

Reply:
So, ideally when you copy private key from a wallet, by default it doesnt have `0x` prefix. So, either you have to add it manually when you are using it in the code or the code needs to append `0x` if its not present.